README.md

```diff
@@ -243,6 +243,150 @@ There are many differences between SQL syntax and functions supported by differe
+exports a number of unsafe functions to call any function, operator or value. This functions
+can be found on Database.Esqueleto.Internal.Sql module.
+
+Warning: the functions discuss on section should always be use with an explicit type signature
+as they are too general in order to maintain type safety, and you will need to indicate some
+type when you want to get results back from the database so better set types earlier rather 
+than late. This functions are also vulnerable to sql injection attacks so their use must be
+limited and with care, more on this later.
+
+The most common use of this functions is for calling RDBMS specific or custom functions,
+for that end we use `unsafeSqlFunction`. For example, if we wish to consult the postgres
+`now` function we could so as follow:
+
+```haskell
+postgresTime :: (MonadIO m, MonadLogger m) => SqlWriteT m UTCTime
+postgresTime = 
+  select (return now) >>= return . head
+  where
+    now :: SqlExpr (Value UTCTime) 
+    now = unsafeSqlFunction (fromText "now") ([] :: [SqlExpr (Value Int)])
+```
+
+which generates this SQL:
+
+```sql
+SELECT now()
+```
+
+With the `now` function we could now use the current time of the postgres RDBMS on any query.
+Do notice that `now` does not use any arguments, so we use an empty list cast to a valid
+`UnsafeSqlFunctionArgument` instance to represent no arguments. 
+
+Note: make () an instance of `UnsafeSqlFunctionArgument` for when no arguments are needed, this
+casting is quite dumb.
+
+We can also use `unsafeSqlFunction` for more complex functions with customs values using 
+`unsafeSqlValue` which turns any string into a sql value of whatever type we want, disclaimer:
+if you use it badly you will cause a runtime error. For example, say we want to try postgres'
+`date_part` function and get the day of a timestamp, we could use:
+
+```haskell
+postgresTimestampDay :: (MonadIO m, MonadLogger m) => SqlWriteT m Int
+postgresTimestampDay = 
+  select (return $ dayPart date) >>= return . head
+  where
+    dayPart :: SqlExpr (Value UTCTime) -> SqlExpr (Value Int) 
+    dayPart s = unsafeSqlFunction (fromText "date_part") (unsafeSqlValue (fromText "\'day\'") :: SqlExpr (Value String) ,s)
+    date :: SqlExpr (Value UTCTime)
+    date = unsafeSqlValue . fromText $ "TIMESTAMP \'2001-02-16 20:38:40\'"
+```
+
+which generates this SQL:
+
+```sql
+SELECT date_part('day', TIMESTAMP '2001-02-16 20:38:40')
+```
+
+Using `unsafeSqlValue` we were able to also define the type of the value.
+
+Another useful unsafe function is `unsafeSqlCastAs`, which allows us to cast any type
+to another within a query. For example, say we want to use our previews `dayPart` function
+on the current system time, we could:
+
+```haskell
+postgresTimestampDay :: (MonadIO m, MonadLogger m) => SqlWriteT m Int
+postgresTimestampDay = 
+  liftIO getCurrentTime >>= \t -> select (return $ dayPart (toTIMESTAMP $ val t)) >>= return . head
```

Comment:
I would really prefer to avoid operators where they make the code less readable. `do` notation would read very nicely here, IMO.